### PR TITLE
integrate with login_dev

### PR DESCRIPTION
**Pull request by rzdao**
integrated with login_dev

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -3,36 +3,46 @@
...
+login_manager.login_view = "/"
+login_manager.login_message = u"you must login to access this page"
+login_manager.login_message_category = "message"
+
 
+# db
+db.init_app(app)
 db.create_all()
+# TODO using a local db.session might be a better idea
 db_manager = DBManager(db.session)
```
Comment (jrdbuch) [inline app.py:30]: I think this global db.session is ok??? FlaskLogin does a bunch of scoped local session under the hood

File: app.py [hunk 2]
```diff
@@ -134,6 +147,7 @@ def login():
 def register():
     """Register a user."""
     data = request.form
+    # TODO we might need also to check if data has username or password
```
Comment (jrdbuch) [inline app.py:150]: if either the password or username is not set correctly the add_user call will return with an error, which is handled gracefully.

File: app.py [hunk 3]
```diff
@@ -152,6 +166,44 @@ def render_login():
...
+@app.route('/test/register/<username>/<password>')
+def test_register(username, password):
+    """Route for test register."""
+    user = User(username=username, hashed_password=int(password))
+    scoped_session = db.create_scoped_session()
+    res = db_manager.DBManager(scoped_session).add_user(user)
+
+    if res.is_ok():
+        return "you are registered {}".format(user.username), 200
+    return "{}".format(res), 400
```
Comment (jrdbuch) [inline app.py:186]: do we need these test endpoints? We should add unit tests that exercise the regular endpoints instead

**Review by tschelbs18 — APPROVED** [review]
Approved per team review

**Pull request merged**